pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -298,17 +309,25 @@ func (r *OAMApplicationReconciler) Reconcile(req reconcile.Request) (result reco
 	r.record.Event(ac, event.Normal(reasonRenderComponents, "Successfully rendered components", "workloads", strconv.Itoa(len(workloads))))
 
 	applyOpts := []resource.ApplyOption{resource.MustBeControllableBy(ac.GetUID())}
-	if r.applyOnceOnly {
+	applyingWorkloads := make([]Workload, len(workloads))
+	copy(applyingWorkloads, workloads)
+	if r.applyOnceOnlyForce {
```

Comment:
the apply function takes the option, we can do the filter there

Reply:
Yes, I will refine this PR after #848